lib/core/test/unit/Cardano/Wallet/DB/Properties.hs

```diff
@@ -862,67 +896,173 @@ prop_rollbackTxHistory db@DBLayer{..} (InitialCheckpoint cp0) (GenTxHistory txs0
     :: GenSparseCheckpointsArgs
     -> Property
-prop_sparseCheckpointTipAlwaysThere (GenSparseCheckpointsArgs (k, h)) = prop
+prop_sparseCheckpointTipAlwaysThere (GenSparseCheckpointsArgs cfg (k, h)) = prop
     & counterexample ("Checkpoints: " <> show cps)
     & counterexample ("h=" <> show h)
     & counterexample ("k=" <> show k)
   where
-    cps = sparseCheckpoints (Quantity k) (Quantity h)
+    cps = sparseCheckpoints cfg (Quantity k) (Quantity h)
 
     prop :: Property
     prop = property $ fromIntegral h `elem` cps
 
--- | Check that sparseCheckpoints always return at least 10 checkpoints (or
--- exactly the current height if h < 10).
+-- | Check that sparseCheckpoints always return at least edgeSize checkpoints (or
+-- exactly the current height if h < edgeSize).
 prop_sparseCheckpointMinimum
     :: GenSparseCheckpointsArgs
     -> Property
-prop_sparseCheckpointMinimum (GenSparseCheckpointsArgs (k, h)) = prop
+prop_sparseCheckpointMinimum (GenSparseCheckpointsArgs cfg (k, h)) = prop
     & counterexample ("Checkpoints: " <> show cps)
     & counterexample ("h=" <> show h)
     & counterexample ("k=" <> show k)
   where
-    cps = sparseCheckpoints (Quantity k) (Quantity h)
+    cps = sparseCheckpoints cfg (Quantity k) (Quantity h)
 
     prop :: Property
-    prop = property $ fromIntegral (length cps) >= min 10 h
+    prop = property $ fromIntegral (length cps) >= min (edgeSize cfg) h
 
 
 -- | Check that sparseCheckpoints always return checkpoints that can cover
 -- rollbacks up to `k` in the past. This means that, if the current block height
 -- is #3000, and `k=2160`, we should be able to rollback to #840. Since we make
--- checkpoints every 100 blocks, it means that block #800 should be in the list.
+-- checkpoints every gapsSize blocks, it means that block #800 should be in the list.
 --
 -- Note: The initial checkpoint at #0 will always be present.
 prop_sparseCheckpointNoOlderThanK
     :: GenSparseCheckpointsArgs
     -> Property
-prop_sparseCheckpointNoOlderThanK (GenSparseCheckpointsArgs (k, h)) = prop
+prop_sparseCheckpointNoOlderThanK (GenSparseCheckpointsArgs cfg (k, h)) = prop
     & counterexample ("Checkpoints: " <> show ((\cp -> (age cp, cp)) <$> cps))
     & counterexample ("h=" <> show h)
     & counterexample ("k=" <> show k)
   where
-    cps = sparseCheckpoints (Quantity k) (Quantity h)
+    cps = sparseCheckpoints cfg (Quantity k) (Quantity h)
 
     prop :: Property
     prop = property $ flip all cps $ \cp ->
-        cp == 0 || (age cp - 100 <= int k)
+        cp == 0 || (age cp - int (gapsSize cfg) <= int k)
 
     age :: Word32 -> Int
     age cp = int h - int cp
 
+-- | This property checks that, the checkpoints kept for an edge size of 0 are
+-- included in the list with a non-null edge size, all else equals.
+prop_sparseCheckpointEdgeSize0
+    :: GenSparseCheckpointsArgs
+    -> Property
+prop_sparseCheckpointEdgeSize0 (GenSparseCheckpointsArgs cfg (k, h)) = prop
```

Comment:
Personally I'm not a fan of separating the actual `prop` and putting it last, and would have done something more like
```haskell
    let cps  = sparseCheckpoints cfg (Quantity k) (Quantity h)
    let cps' = sparseCheckpoints (cfg { edgeSize = 0 }) (Quantity k) (Quantity h)
    property (cps' `L.isSubsequenceOf` cps)
        & counterexample ("Checkpoints: " <> show cps)
        & counterexample ("h=" <> show h)
        & counterexample ("k=" <> show k)
```
for easier reading from top to bottom.

But I guess this style is common in this module.